lib/postgrex/notifications.ex

```diff
@@ -188,38 +214,39 @@ defmodule Postgrex.Notifications do
   ## CALLBACKS ##
 
   @doc false
-  def init(opts) do
-    idle_timeout = opts[:idle_timeout]
-
-    if idle_timeout do
-      require Logger
-
-      Logger.warn(
-        ":idle_timeout in Postgrex.Notifications is deprecated, " <>
-          "please use :idle_interval instead"
-      )
-    end
+  def init({mod, args, opts}) do
+    with {:ok, mod_state} <- mod.init(args) do
```

Comment:
I would explicitly check that it has to return {:error, _} or :ignore otherwise, instead of relying on connection.